### PR TITLE
Change ads provider, and add consent system for advertising

### PR DESCRIPTION
**Pull request by Redblueflame**
What's left to do:
- [ ] Add consent withdrawal support for analytics (unami)

### CODE REVIEW

**Comment by Redblueflame** [comment]
@Geometrically Everything should be ready for review.

File: modules/analytics/lib/templates/plugin.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    // Rely on the client
+    cookies = document.cookie;
+  }
+  let processed = {}
+  cookies.split(';').forEach((e) => {
+    let val = e.trim().split('=');
+    processed[val[0]] = decodeURI(val[1]);
+  })
+  let scopes = decodeURIComponent(processed['modrinth-scopes']).split(",");
+  return (scopes !== null && scopes.includes('analytics'));
```
Comment (Geometrically) [inline modules/analytics/lib/templates/plugin.js:18]: I would use `nuxt-cookies-universal` for this. It's a lot cleaner and the auth already uses it.

File: pages/dashboard.vue [hunk 2]
```diff
@@ -2,7 +2,7 @@
   <div class="page-container">
     <div class="page-contents">
       <div class="sidebar-l">
-        <div class="card page-nav">
+        <div v-if="$auth.user != null" class="card page-nav">
```
Comment (Geometrically) [inline pages/dashboard.vue:5]: Why not just `$auth.user`

File: components/ads/Advertisement.vue [hunk 3]
```diff
@@ -1,34 +1,36 @@
...
     },
-    pageUrl: {
+    adUnit: {
       type: String,
-      required: false,
-      default: '',
+      required: true,
     },
   },
+  mounted() {},
```
Comment (Geometrically) [inline components/ads/Advertisement.vue:23]: Why the empty mounted hook?

**Review by Geometrically — CHANGES_REQUESTED** [review]
Looks good! Just a couple minor changes, thank you so much :+1:

File: modules/analytics/lib/templates/plugin.js (hunk 1 above)
Comment (Redblueflame) [reply modules/analytics/lib/templates/plugin.js:18]: Unfortunatly, it's in a module, so dependencies are not allowed there

File: pages/dashboard.vue (hunk 2 above)
Comment (Redblueflame) [reply pages/dashboard.vue:5]: Just to explain what it does, having a inherent false if empty is just too cryptic to be in there

File: components/ads/Advertisement.vue (hunk 3 above)
Comment (Redblueflame) [reply components/ads/Advertisement.vue:23]: Let me remove that

**Review by Geometrically — APPROVED** [review]

**Pull request merged**